hail/python/hailtop/utils/utils.py

```diff
@@ -862,8 +862,10 @@ async def periodically_call(period, f, *args, **kwargs):
     async def loop():
         log.info(f'starting loop for {f.__name__}')
         while True:
-            await f(*args, **kwargs)
-            await asyncio.sleep(period)
+            try:
+                await f(*args, **kwargs)
+            finally:
+                await asyncio.sleep(period)
```

Comment:
This means there's a delay before we see the error message. retry_long_running already has a delay built in. Do we need to make that more configurable?

Reply:
@daniel-goldstein I thought this could be where the problem was with using a ton more disk space than expected if we reach an error and are constantly writing to the file with no sleep in between. I might be wrong that this is true and whether this is the function that needs to change.

The delay / sleep in retry_long_running is a different delay than the periodically call delay. For example, the sleep we want is 5 seconds while retry_long_running has a sleep backoff starting at 0.1 seconds up to 30 seconds.

Reply:
I think I would prefer here that we push the try/except into the `f` that we're passing to periodically_call. If we expect it to raise and don't want it to, we can silence those errors closer to the source and then not change the periodically_call semantics.